Framework/src/DataDump.cxx

```diff
@@ -0,0 +1,295 @@
+
+    static int representation = 0, old_representation = 1;
+    old_representation = representation;
+    ImGui::RadioButton("hexadecimal", &representation, 0);
+    ImGui::SameLine();
+    ImGui::RadioButton("binary", &representation, 1);
+
+    // scrollable area
+    ImGui::BeginChild("##ScrollingRegion", ImVec2(0, ImGui::GetFontSize() * 25), false);
+    // table
+    ImGui::Columns(5, "payload_display", true);
+
+    // header row
+    ImGui::Separator();
+    resizeColumns(representation, old_representation);
+    ImGui::Text("");
+    ImGui::NextColumn();
+    ImGui::Text("#1");
+    ImGui::NextColumn();
+    ImGui::Text("#2");
+    ImGui::NextColumn();
+    ImGui::Text("#3");
+    ImGui::NextColumn();
+    ImGui::Text("#4");
+    ImGui::NextColumn();
+    ImGui::Separator();
+
+    // print the hex values in the columns and rows of the table
+    vector<string> formattedData =
+      (representation == 0)
+        ? getHexRepresentation(DataDump::guiState.current_payload.data, DataDump::guiState.current_payload.size)
+        : getBinRepresentation(DataDump::guiState.current_payload.data, DataDump::guiState.current_payload.size);
+    int line = 0;
+    static int selected = -1;
+    for (unsigned long pos = 0; pos < formattedData.size();) {
+      char label[32];
+      sprintf(label, "%04d", line * 4);
+      if (ImGui::Selectable(label, selected == line, ImGuiSelectableFlags_SpanAllColumns)) {
+        selected = line;
+      }
+      for (int i = 0; i < 4; i++) { // four columns
+        ImGui::NextColumn();
+        string temp;
+        for (int j = 0; j < 2; j++) { // 2 x 16 bits words (char)
+          if (pos < formattedData.size()) {
+            temp += formattedData[pos];
+            pos++;
+          }
+        }
+        ImGui::TextUnformatted(temp.c_str());
+      }
+      ImGui::NextColumn();
+      line++;
+    }
+
+    // footer
+    ImGui::Columns(1);
+    ImGui::EndChild();
+    ImGui::Separator();
+  }
+}
+
+void updateHeaderGui()
+{
+  if (DataDump::guiState.current_header.data == nullptr) {
+    ImGui::Text("No data loaded yet, click Next.");
+  } else { // all the stuff below should go to a method
+    auto* header = header::get<header::DataHeader*>(DataDump::guiState.current_header.data);
+    ImGui::Text("sMagicString : %d", o2::header::DataHeader::sMagicString);
+    ImGui::Text("sVersion : %d", o2::header::DataHeader::sVersion);
+    ImGui::Text("sHeaderType : %s", o2::header::DataHeader::sHeaderType.as<string>().c_str());
+    ImGui::Text("sSerializationMethod : %s", o2::header::DataHeader::sSerializationMethod.as<string>().c_str());
+
+    ImGui::Text("Header size : %d", header->headerSize);
+    ImGui::Text("Payload size : %ld", header->payloadSize);
+    ImGui::Text("Header version : %d", header->headerVersion);
+    ImGui::Text("flagsNextHeader : %d", header->flagsNextHeader);
+    ImGui::Text("description : %s", header->description.as<string>().c_str());
```

Comment:
It might seem to be the same, but `description` is something different than `dataDescription`, which keeps the actual description of the data. Also, you might want to print `dataOrigin`. However, both of these might not be null-terminated if the string size equals maximum capacity.
You can also access the string version by `header->description.str`

Reply:
done